src/test/java/bio/terra/app/controller/DatasetsApiControllerTest.java

```diff
@@ -438,4 +441,35 @@ private static MockHttpServletRequestBuilder getRequest(String tableName, String
         .queryParam("direction", DIRECTION.name())
         .queryParam("filter", FILTER);
   }
+
+  @Test
+  void lockDataset() throws Exception {
+    var fakeFlightId = "fakeFlightId";
+    when(datasetService.manualExclusiveLock(TEST_USER, DATASET_ID)).thenReturn(fakeFlightId);
+    when(jobService.retrieveJob(fakeFlightId, TEST_USER))
+        .thenReturn(new JobModel().id(fakeFlightId));
+    mockValidators();
+
+    mvc.perform(put(LOCK_DATASET_ENDPOINT, DATASET_ID))
+        .andExpect(status().is2xxSuccessful())
+        .andReturn();
```

Comment:
Do we care to test what's returned here?

Reply:
I was thinking that it wasn't too important since it would just be a job result, but yes, definitely now that it's returning the ResourceLocks object. I'll add these checks!